test/units/module_utils/test_api.py

```diff
@@ -46,3 +45,53 @@ def login_database():
 
         with pytest.raises(Exception):
             login_database()
+
+    def test_no_retries(self):
+
+        @retry()
+        def login_database():
+            assert False, "Should not execute"
+
+        login_database()
+
+
+class TestRetryWithDelaysAndCondition:
+
+    def test_empty_retry_iterator(self):
+        @retry_with_delays_and_condition(backoff_iterator=[])
+        def login_database():
+            login_database.counter += 1
+
+        login_database.counter = 0
+        r = login_database()
+        assert login_database.counter == 1
+
+    def test_no_retry_exception(self):
+        @retry_with_delays_and_condition(
+            backoff_iterator=[1],
+            should_retry_error=lambda x: False,
+        )
+        def login_database():
+            login_database.counter += 1
+            if login_database.counter == 1:
+                raise Exception("Error")
+
+        login_database.counter = 0
+        with pytest.raises(Exception):
```

Comment:
Always use raises with `match=` or you'll catch false positives. Especially because almost any exception is a subclass of `Exception`.
```suggestion
        with pytest.raises(Exception, match='Error'):
```

Reply:
Honestly, I'd change it to some `class CustomTestException(Exception): ...` to avoid any weird corner cases.